core/GlobalState.cc

```diff
@@ -911,13 +911,13 @@ NameRef GlobalState::enterNameUTF8(string_view nm) {
 
 NameRef GlobalState::enterNameConstant(NameRef original) {
     ENFORCE(original.exists(), "making a constant name over non-existing name");
-    ENFORCE(original.data(*this)->kind == UTF8 ||
-                (original.data(*this)->kind == UNIQUE &&
+    ENFORCE(original.data(*this)->kind == NameKind::UTF8 ||
```

Comment:
if we make NameKind as enum class we'll be forced to you this syntax, and thus enforcing consistency.
Do you want to try that?